kie-server-parent/kie-server-services/kie-server-services-kafka/src/main/java/org/kie/server/services/jbpm/kafka/KafkaServerExtension.java

```diff
@@ -140,6 +159,10 @@ public void destroy(KieServerImpl kieServer, KieServerRegistry registry) {
         if (deploymentService != null) {
             deploymentService.removeListener(this);
         }
+        if (producerReady.compareAndSet(true, false)) {
+            producer.close();
```

Comment:
also, shouldn't it be beneficial to call `close` with a timeout? maybe configurable through a system property - wdyt?

Reply:
I would normally agree, but this is destroy, which is invoked only when server is closed, default timeout wil be fine, we do not want to give the user to specify a long one and interfere on the server cleanup procedure

Reply:
then, maybe we should actually invoke the `close(Duration timeout)` instead passing a low value cause the default timeout for the `close()` method is pretty high (it's Long.MAX_VALUE):

https://github.com/apache/kafka/blob/2.6/clients/src/main/java/org/apache/kafka/clients/producer/KafkaProducer.java#L1158

Reply:
done